### DISCUSSION TITLE
Some classes stopped working

### DISCUSSION THREAD

**Discussion by Niklas-pixel** [tailwindlabs/tailwindcss#11333] [Help]
Hi if the is anyone out there :)
All my CSS worked when i first pushed it to github, and then the padding and hover on the buttons stopped working.

Button component:

import Link from "next/link"
import { IconType } from "react-icons"

type ButtonProps = {
    innerText?: string,
    bgColor: string,
    hBgColor: string,
    icon?: IconType,
    px?: number,
    py?: number,
    children?: React.ReactNode | string
    path?: string
}
 
export const LinkButton = ({path, px, py, innerText, bgColor, hBgColor, icon: Icon }: ButtonProps ) => {
  return (
      <Link className="inline-block" href={`${path}`}>
         <button   className={`hover:${hBgColor} ${bgColor} py-${py} px-${px} text-white ${bgColor === "bg-red-700" && `shadow-lg shadow-red-500/50`} `}>
         {innerText} {Icon && <Icon className="inline-block" />}
        </button></Link>   
  )
}
 

tailwind.config.js

`/** @type {import('tailwindcss').Config} */
module.exports = { mode: 'jit',
// These paths are just examples, customize them to match your project structure
purge: [
  './public/**/*.html',
  './src/**/*.{js,jsx,ts,tsx}',
],
  content: [
    './src/pages/**/*.{js,ts,jsx,tsx,mdx}',
    './src/components/**/*.{js,ts,jsx,tsx}',
    './src/app/**/*.{js,ts,jsx,tsx,mdx}',
  ],
  theme: {
    extend: {
      backgroundImage: {
        'gradient-radial': 'radial-gradient(var(--tw-gradient-stops))',
        'gradient-conic':
          'conic-gradient(from 180deg at 50% 50%, var(--tw-gradient-stops))',
      },
    },
  },
  plugins: [],
}

`

**Comment by wongjn** [comment]
As per [the documentation](https://tailwindcss.com/docs/content-configuration#dynamic-class-names):

> The most important implication of how Tailwind extracts class names is that it will only find classes that exist as _complete unbroken strings_ in your source files.
>
> If you use string interpolation or concatenate partial class names together, Tailwind will not find them and therefore will not generate the corresponding CSS:
>
> **Don’t construct class names dynamically**
>
> ```vue
> <div class="text-{{ error ? 'red' : 'green' }}-600"></div>
> ```
> In the example above, the strings `text-red-600` and `text-green-600` do not exist, so Tailwind will not generate those classes.
> Instead, make sure any class names you’re using exist in full:
>
> **Always use complete class names**
>
> ```vue
> <div class="{{ error ? 'text-red-600' : 'text-green-600' }}"></div>
> ```
